cdk/lib/security-hq.ts

```diff
@@ -1,17 +1,137 @@
+import { Duration, RemovalPolicy } from '@aws-cdk/core';
+import type { App, CfnElement } from '@aws-cdk/core';
+import { AccessScope, GuApplicationPorts, GuEc2App } from '@guardian/cdk';
+import { GuAlarm } from '@guardian/cdk/lib/constructs/cloudwatch';
 import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
-import { GuStack, GuStageParameter } from '@guardian/cdk/lib/constructs/core';
+import {
+  GuDistributionBucketParameter,
+  GuParameter,
+  GuStack,
+} from '@guardian/cdk/lib/constructs/core';
+import {
+  GuDynamoDBReadPolicy,
+  GuDynamoDBWritePolicy,
+} from '@guardian/cdk/lib/constructs/iam';
 
 export class SecurityHQ extends GuStack {
   migratedFromCloudFormation = true;
 
   constructor(scope: App, id: string, props: GuStackProps) {
     super(scope, id, props);
 
-    new CfnInclude(this, `security-hq-PROD`, {
-      templateFile: '../cloudformation/security-hq.template.yaml',
-      parameters: { Stage: GuStageParameter.getInstance(this) },
+    const table = new Table(this, 'DynamoTable', {
+      tableName: `security-hq-iam-${this.stage}`,
+      removalPolicy: RemovalPolicy.RETAIN,
+      readCapacity: 5,
+      writeCapacity: 5,
+      partitionKey: {
+        name: 'id',
+        type: AttributeType.STRING,
+      },
+    });
+
+    // Required as migrated from existing Cloudformation stack.
+    const defaultChild = table.node.defaultChild as unknown as CfnElement;
+    defaultChild.overrideLogicalId('SecurityHqIamDynamoTable');
+
+    const distBucket = GuDistributionBucketParameter.getInstance(this);
+
+    // Create new ALB-based app, that will live in the new VPC. This will run
+    // simultaneously with the old service. Once confident, DNS can be switched
+    // and the old app deleted.
+    new GuEc2App(this, {
+      access: { scope: AccessScope.PUBLIC },
+      app: 'security-hq',
+      applicationPort: GuApplicationPorts.Play,
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.LARGE),
+      certificateProps: {
+        PROD: { domainName: 'security-hq.gutools.co.uk' },
+        CODE: { domainName: 'security-hq.code.dev-gutools.co.uk' }, // Note, Security HQ does not in fact have a CODE stage.
+      },
+      monitoringConfiguration: { noMonitoring: true },
+      scaling: {
+        CODE: { minimumInstances: 1 },
+        PROD: { minimumInstances: 1 },
+      },
+      userData: `
+#!/bin/bash -ev
+# setup security-hq
+mkdir -p /etc/gu
+
+aws --region eu-west-1 s3 cp s3://${distBucket.value.toString()}/security/${
+        this.stage
+      }/security-hq/security-hq.conf /etc/gu
+aws --region eu-west-1 s3 cp s3://${distBucket.value.toString()}/security/${
+        this.stage
+      }/security-hq/security-hq-service-account-cert.json /etc/gu
+aws --region eu-west-1 s3 cp s3://${distBucket.value.toString()}/security/${
+        this.stage
+      }/security-hq/security-hq.deb /tmp/installer.deb
+
+dpkg -i /tmp/installer.deb`,
```

Comment:
Did the GuCDK types not work here?

Reply:
We need to move the config into specific locations and I couldn't see a way to do that with the in-built types. (It's quite possible I've missed that though.) Current thinking was to get this live and then review the user-data approach and refine.